### ISSUE TITLE
Document magic numbers in exit code handling in our test helpers

### ISSUE THREAD

**Issue by deivid-rodriguez** [rubygems/rubygems#5165]
We use a some magic number when dealing with exit codes in our help that runs subprocesses via `Open3.popen3`.

Here: https://github.com/rubygems/rubygems/blob/f62cad4818cc333b40d506d859ca53531f2f25f6/bundler/spec/support/subprocess.rb#L39-L54

We should probably add a link to https://tldp.org/LDP/abs/html/exitcodes.html and move it to a constant/variable with some non magical name.

_Originally posted by @doodzik in https://github.com/rubygems/rubygems/pull/4992/files#r734699098_

**Comment by gustavothecoder** [comment]
Hi! I'd like to work on this issue

**Comment by deivid-rodriguez** [comment]
Great! Hopefully with the link above it should be easier to understand why we do this and document it. I honestly don't remember myself the exact situation that led me to change this, which proves the value of better documenting this 😅.

**Comment by gustavothecoder** [comment]
@deivid-rodriguez do you think it's a good idea to create a file to centralize these codes? Then we can use them in other files later.

I'm thinking about something like this:
```Ruby
module ExitCodes
  # For details see: https://tldp.org/LDP/abs/html/exitcodes.html
  GENERAL_ERROR = 1
  MISUSE_OF_SHELL_BUILTINS = 2
  COMMAND_INVOKED_CANNOT_EXECUTE = 126
  COMMAND_NOT_FOUND = 127
  INVALID_ARGUMENT_TO_EXIT = 128
  FATAL_ERROR_SIGNAL = 128
  SCRIPT_TERMINATED_BY_CONTROL_C = 130
  EXIT_STATUS_OUT_OF_RANGE = 255
end
```

**Comment by deivid-rodriguez** [comment]
@gustavothecoder Thanks for compiling that constant list. I believe though, this is a very niche concern very specific to our specs, and we won't need it anywhere else, so it feels overkill to add code that we most likely will never use.

I think the basic idea here is to document the 128 magic number by explaining rows 6 and 7 in the table at https://tldp.org/LDP/abs/html/exitcodes.html. That if the subprocess exists because of a signal, then the exit code set by bash is `128` plus the signal number, so we try to "simulate" that for a process run by `Open3.popen3`.

However, I'm not really sure whether we need this at all, given that it's bash specific, and Bundler specs can be run on different shells. I introduced this at 814e293e23fbb9452fa23bb5aa42389ec5e5d731, but I failed to explain the kind of issues this was creating. Maybe we should revert it, I'm not sure.
